grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/binary/RelationshipAtom.java

```diff
@@ -442,6 +442,7 @@ else if(compatibleTypesFromRoles.isEmpty() || types.isEmpty()){
 
         return compatibleTypes.asMap().entrySet().stream()
                 .sorted(Comparator.comparing(e -> -e.getValue().size()))
+                .sorted(Comparator.comparing(e -> e.getKey().isImplicit()))
```

Comment:
Can we add a comment explaining why the ordering is necessary? I assume it's either some technical reason or just to make it easier to read.

Reply:
it's the order in which the relations are looked at, so trying to prioritise the ones that most probably will yield answers